utils/time_utils.py

```diff
@@ -0,0 +1,32 @@
+import time
+from utils.print_utils import print_distributed
+
+
+class TimerError(Exception):
+    """A custom exception used to report errors in use of Timer class"""
+
+
+class Timer:
+    def __init__(self, code_section_monitored: str, verbosity_level: int):
+        self.start_time = None
+        self.code_section_monitored = code_section_monitored
+        self.verbosity_level = verbosity_level
+
+    def start(self):
+        """Start a new timer"""
+        if self.start_time is not None:
+            raise TimerError(f"Timer is running. Use .stop() to stop it")
+
+        self.start_time = time.perf_counter()
+
+    def stop(self):
+        """Stop the timer, and report the elapsed time"""
+        if self.start_time is None:
+            raise TimerError(f"Timer is not running. Use .start() to start it")
+
+        elapsed_time = time.perf_counter() - self.start_time
```

Comment:
Should we store a dictionary of times? We could then access all the performance details at any point